caddyhttp/proxy/reverseproxy.go

```diff
@@ -126,6 +127,40 @@ func NewSingleHostReverseProxy(target *url.URL, without string) *ReverseProxy {
 	return rp
 }
 
+func NewDynamicHostReverseProxy(templateUrl string, without string) *ReverseProxy {
+	templateUrl = strings.Replace(strings.ToLower(templateUrl), "{host}", "HOST", -1)
+	target, err := url.Parse(templateUrl)
+	if err != nil {
+		return nil
+	}
+	targetQuery := target.RawQuery
+	director := func(req *http.Request) {
+		replacer := httpserver.NewReplacer(req, nil, "")
+
+		req.URL.Scheme = target.Scheme
+		req.URL.Host = replacer.Replace(strings.Replace(target.Host, "HOST", "{host}", -1))
```

Comment:
That's why the ToLower but that might also cause issues on second thought. url.Parse seems to allow many special characters but {}